app/jobs/award_badge_job.rb

```diff
@@ -17,7 +17,7 @@ def self.worth_queuing?(badge_slug, context:)
     return true if context.blank?
 
     context_key = context.class.base_class.name.demodulize.underscore
-    badge = "Badges::#{badge_slug.to_s.camelize}Badge".constantize
+    badge = "Badges::#{badge_slug.to_s.camelize}Badge".safe_constantize
```

Comment:
@SleeplessByte I've opted for the `safe_constantize` option, as the `badge.worth_queuing?` call on the next line will then fail with a nil reference error, which we'll automatically log.